### ISSUE TITLE
[Q] Notifications when sweep finishes

### ISSUE THREAD

**Issue by andreimargeloiu** [wandb/client#3504]
**Is there a way to receive a notification (e.g., email) when a sweep terminates?**

**Context:** I'm using the sweep to set multiple experiments, in which I train one model over 1) multiple datasets and 2) with different sparsity losses. I run the sweep in grid_search mode because I want to know all the results. Then, I wait until the sweep finishes and analyze the results.

**Problem:** I can't estimate exactly when the sweep will finish. Thus, I'm entering the wandb UI multiple times whether the sweep has finished.

**Comment by ramit-wandb** [comment]
Hi @andreimargeloiu,

We do allow for notifications inside runs through the use of [wandb.alert](https://docs.wandb.ai/guides/track/alert). I can see something like the following working:

```python3
sweep_id = wandb.sweep(<Sweep_Config>)
wandb.agent(sweep_id, func=<Your_Function>) # Assuming this is a finite agent, so either runs grid search or has an explicit count = ...

wandb.alert(title='Completed', text='Sweep has been completed')
```

Please let me know if this works for your use case.

Thanks,
Ramit

**Comment by andreimargeloiu** [comment]
I'm creating a sweep via `wand sweep sweep_script.yaml` and then start multiple agents. Each agent executes the sweep, which means multiple agents run the `main.py` script simultaneously. As I understand, if I put the `wandb. alert()` is the main script, then it will trigger an email for any run finished.

How could you make the `wandb.alert` trigger only when the entire sweep has finished?

**Comment by andreimargeloiu** [comment]
@ramit-wandb any updates?

**Comment by ramit-wandb** [comment]
Hi @andreimargeloiu,

I had made a feature request for this, but currently the only way to do this currently is the method I suggested above. The method I suggested above will not trigger after every run, but only after `wandb.agent` has completed all the runs.

I do foresee an issue if you have multiple agents running simultaneously where each agent will send an individual alert. The only solution I see around that right now is to add a `wait_duration` parameter to your `alert`, which will wait the number of seconds passed before sending another alert.

Thanks,
Ramit
